readthedocs/search/parse_json.py

```diff
@@ -93,6 +109,30 @@ def _is_section(tag):
     )
 
 
+def _parse_code_section(tag):
+    """
+    Parse a code section to fetch relevant content only.
+
+    Sphinx has ``pre`` tags within a div with a ``highlight`` class.
+    Other ``pre`` tags are used for line numbers (we don't index those).
```

Comment:
This feels pretty brittle, does it work on html4 and html5 builders? I'd prefer a more generic solution that does something like turn the codeblock into text, removing all html tags, and stripping initial numbers or something. This feels likely to break and very specific :/

Reply:
So, sphinx puts the lines and code in separated pre tags, this will be extracted like `1\n2\n3\n4 ... code` instead of `1 code\n2 ...`, that's why we can't strip the numbers. I looked into the generated code for mkdocs code blocks, they just use a `pre` tag (I couldn't see any example with line numbers). I also want to refactor all these functions into a class, I'm not sure if we can have something generic enough for all static generators, but we can at least share some code.

We could also just see if the rendered text are all consecutive numbers, but that's feel more fragile

Reply:
Found a mkdocs project with line numbers, they  use tables as well

```html
<tbody><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8
9</pre></div></td><td class="code">
<div class="codehilite">
<pre>
<code>
...
</code></pre></div>
</td></tr></tbody>
```

I think I can try removing nodes with the `linenos` or `linenodiv` classes.